compiler/rustc_metadata/src/locator.rs

```diff
@@ -721,28 +721,33 @@ impl<'a> CrateLocator<'a> {
                 return Err(CrateError::ExternLocationNotFile(self.crate_name, loc.clone()));
             };
 
-            if file.starts_with("lib") && (file.ends_with(".rlib") || file.ends_with(".rmeta"))
-                || file.starts_with(self.target.dll_prefix.as_ref())
-                    && file.ends_with(self.target.dll_suffix.as_ref())
-            {
+            match file.starts_with("lib").then_some(()).and_then(|()| {
```

Comment:
? isn't this just `.then`

Reply:
No, because the closure still needs to return `Option<T>` rather than `T`.

But this can be `.then(...).flatten()`. Making that change causes rustfmt to increase indentation, so I've left it as is.

Reply:
To be honest I find this `.then_some(()).and_then(|()| {` very hard to read. Can't we just do something straightforward like https://github.com/jieyouxu/rust/commit/6eba971bde7c2a05448e0073f0da100a145c4568?

Reply:
Thanks for the suggestion. I was trying to avoid repeating conditions (your code repeats `file_name.starts_with("lib")` twice).

I replaced this with a more imperative closure which should be easier to understand. Can you take a look?

Reply:
Frankly, I don't think it's worth avoiding the trivial repetitions, I still find this hard to follow than just doing the simple thing. I let other compiler reviewers decide what they find more readable.

Reply:
Would you mind removing the waiting-on-author label?

Reply:
> Would you mind removing the waiting-on-author label?

You can write `@rustbot ready` to indicate that.